src/index.js

```diff
@@ -135,17 +138,20 @@ class Uploader {
                         result[id].raw = response;
                     }
 
-                    this.updateFileObject(id, 'loaded', true);
-                    this.updateFileObject(id, 'onLoadResponse', result[id]);
+                    this.updateFileMeta(id, 'loaded', true);
+                    this.updateFileMeta(id, 'onLoadResponse', result[id]);
             
                     this.onLoad && this.onLoad(result, this.getFiles());
+                })
+                .catch((error) => {
+                    this.onError && this.onError(Object.assign({ id: id }, error));
```

Comment:
1- I would set a default value for this.onError (empty function) so you don't have toe check it
2- what is the interface of the object? 
the id is the attachment name from outside so I would return 
```
{
attachmentName: ...,
error: ...
}
``